### PR TITLE
Mgbbs/hidden for checkbox render mode

### PR DESCRIPTION
**Pull request by mgbbs**
Summary of the changes
 - Added configurable CheckBox hidden input render mode

Addresses #12833

### CODE REVIEW

**Comment by mgbbs** [comment]
@Eilon is there an area with resources to help resolve build issues?  I'm seeing an npm failure trying to resolve @mirosoft/signalr package, plus a few xunit tests are failing but I can't view the log details.  All of the errors are unrelated to any of the changes in this pull request.  I've tried some literal searches but nothing useful has turned up.

**Comment by Eilon** [comment]
Hi @mgbbs I actually don't work on this project anymore, I just help with labeling issues 😄 

I'm tagging @aspnet/build and @mkArtakMSFT who should be able to help more with the build issues you're seeing (as well as reviewing the PR).

**Comment by pranavkm** [comment]
@mgbbs @BrennanConroy fixed the npm issue fairly recently. I think rebasing on master would help.
The test failures are a known issues and can be ignored (they're marked as flaky and will not fail the build).

**Comment by BrennanConroy** [comment]
The fix still needs to be merged to master which could be later today or tomorrow

**Comment by mgbbs** [comment]
Thanks guys, I assume it's commit e5485e987b so I'll watch for that to hit master then rebase.

File: src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs
Comment (dougbu) [inline src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs]: Suggest this should be in the same namespace as `HtmlHelperOptions`: `Microsoft.AspNetCore.Mvc.ViewFeatures` but defer to @pranavkm here

File: src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs
Comment (dougbu) [inline src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs]: Comment should be more specific. Mention rendering at the end of the form element is _preferred_ but the hidden input field will be rendered inline with the checkbox if `!ViewContext.FormContext.CanRenderAtEndOfForm`.

Comment (dougbu) [reply src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs]: Also, please make sure tests cover this fallback to `Inline` behaviour for both HTML and tag helpers. (The code looks correct but test to keep it that way.)

File: src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs
Comment (mgbbs) [reply src/Mvc/Mvc.ViewFeatures/src/Rendering/CheckBoxHiddenInputRenderMode.cs]: Thanks for the quick response @dougbu. I initially put it in the `.Rendering` namespace to follow suit with `Html5DateRenderingMode` and `TagRenderMode`, but will also defer to @pranavkm.

**Comment by mgbbs** [comment]
@pranavkm this is rebased and updated per @dougbu's comments, and built successfully.

**Comment by dougbu** [comment]
Looks good to me. Over to you @pranavkm, especially on the namespace choice for the new `enum`

**Comment by mkArtakMSFT** [comment]
@dougbu please sign off, if you're done and happy with it. Let's get @pranavkm 's signoff too then to merge this.

**Review by dougbu — APPROVED** [review]

**Review by pranavkm — APPROVED** [review]
Thanks for your PR @mgbbs

**Comment by Tabgyn** [comment]
Is it available in .NET Core 3.1? How do I set this enumerator for all my application?

**Comment by mgbbs** [comment]
> 
> 
> Is it available in .NET Core 3.1? How do I set this enumerator for all my application?

@Tabgyn looks like it'll be available in .NET 5.  It will be configured like so:
`````
            services.Configure<MvcViewOptions>(options =>
            {
                options.HtmlHelperOptions.CheckBoxHiddenInputRenderMode = CheckBoxHiddenInputRenderMode.Inline;
            });

**Pull request merged**
